### PR TITLE
deps: upgrade ed25519-dalek

### PR DESCRIPTION
**Pull request by divagant-martian**
## Change checklist

- [ ] Self-review.
- [ ] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

**Review by dignifiedquire — DISMISSED** [review]

**Comment by dignifiedquire** [comment]
sth is fishy, all cli tests are failing

**Comment by divagant-martian** [comment]
yep.. looking at how to fix it, not sure how to yet. Tried a couple of things without luck

**Comment by divagant-martian** [comment]
actually if I try this on `main` I get a failure too :thinking:

File: iroh-net/src/key/node.rs [hunk 1]
```diff
@@ -106,7 +106,7 @@ pub struct SecretKey(crypto_box::SecretKey);
 
 impl From<crate::tls::SecretKey> for SecretKey {
     fn from(key: crate::tls::SecretKey) -> Self {
-        SecretKey(crypto_box::SecretKey::from(key.to_scalar_bytes()))
+        SecretKey(crypto_box::SecretKey::from(key.to_scalar()))
```
Comment (divagant-martian) [inline iroh-net/src/key/node.rs:109]: this is depending on https://docs.rs/crypto_box/0.9.0/crypto_box/struct.SecretKey.html#impl-From%3CScalar%3E-for-SecretKey
so I'm not sure what's wrong

**Comment by flub** [comment]
Please update to remove the cargo-deny exception in #1361

**Comment by dignifiedquire** [comment]
closing in favor of #1373

**Pull request closed without merging**